packages/schematics/package-update/utility/npm.ts

```diff
@@ -0,0 +1,266 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import { JsonObject, logging } from '@angular-devkit/core';
+import {
+  Rule,
+  SchematicContext,
+  SchematicsException,
+  Tree,
+  chain,
+} from '@angular-devkit/schematics';
+import * as http from 'http';
+import { Observable } from 'rxjs/Observable';
+import { ReplaySubject } from 'rxjs/ReplaySubject';
+import { concat, ignoreElements, map, mergeMap } from 'rxjs/operators';
+import * as semver from 'semver';
+
+const semverIntersect = require('semver-intersect');
+
+const kPackageJsonDependencyFields = [
+  'dependencies',
+  'devDependencies',
+  'peerDependencies',
+  'optionalDependencies',
+];
+
+
+const npmPackageJsonCache = new Map<string, Observable<JsonObject>>();
+
+function _getVersionFromNpmPackage(json: JsonObject, version: string, loose: boolean): string {
+  const distTags = json['dist-tags'] as JsonObject;
+  if (distTags && distTags[version]) {
+    return (loose ? '^' : '') + distTags[version] as string;
```

Comment:
Why is `^` used for `loose` here but `~` below?

Reply:
Changed back to that.